src/libraries/Microsoft.Extensions.Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -9,13 +9,13 @@ namespace Microsoft.Extensions.Hosting.Internal
 {
     internal static class HostingLoggerExtensions
     {
-        public static void ApplicationError(this ILogger logger, EventId eventId, string message, Exception exception)
+        public static void ApplicationError(this ILogger logger, EventId eventId, string? message, Exception? exception)
         {
             if (exception is ReflectionTypeLoadException reflectionTypeLoadException)
             {
-                foreach (Exception ex in reflectionTypeLoadException.LoaderExceptions)
+                foreach (Exception? ex in reflectionTypeLoadException.LoaderExceptions)
                 {
-                    message = message + Environment.NewLine + ex.Message;
+                    message = message + Environment.NewLine + ex!.Message;
```

Comment:
When the null-forgivin operator `!` is used here, then why is in L16 the exception annoted to allow null?

Either disallow null for the exception or guard against null.